qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/definition/QuteDefinitionInForUserTagTest.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.definition;
+
+import static com.redhat.qute.QuteAssert.ll;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testDefinitionFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test definition with user tag.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class QuteDefinitionInForUserTagTest {
+
+	@Test
+	public void param() throws Exception {
+		String template = "{#for item in items}\r\n" + //
+				"	{#linkItem item name=it|em.name item /}\r\n" + //
+				"{/for}";
+		testDefinitionFor(template, "test.qute", //
+				ll("test.qute", r(1, 22, 1, 26), r(0, 6, 0, 10)));
+	}
+
+	@Test
+	public void it() throws Exception {
+		String template = "{#for item in items}\r\n" + //
+				"	{#linkItem it|em name=item.name item /}\r\n" + //
+				"{/for}";
+		testDefinitionFor(template, "test.qute", //
+				ll("test.qute", r(1, 12, 1, 16), r(0, 6, 0, 10)));
+	}
+
+	@Test
+	public void noIt() throws Exception {

```

Comment:
what do it/noIt mean?

Reply:
See https://quarkus.io/guides/qute-reference#user_tags

Reply:
in user tag you can write parameter without value like

`{#user name}` 

in this case name is the it of the user tag.

You can hav ejust one it, so if you write

`{#user name name2}` 

name2 is not an it (it should have an error report for that)